### PR TITLE
Run integration test against pypi and github dependencies

### PR DESCRIPTION
**Pull request by lichenbo**
I use tox to manage dependency matrix. Use pip freeze to collect the dependency requirement and used in dockerfile to build the docker container. The correct version of the dependencies and colcon-bundle is used when doing integration tests both locally and in travis.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+[testenv:py35-pypi]
+deps =
+    colcon-core
+    colcon-ros-bundle
+commands = ./run_integration_test.sh
+
+[testenv:py35-github]
+deps =
+    -e git+https://github.com/colcon/colcon-core.git#egg=colcon-core
```
Comment (murphm8) [inline tox.ini]: Should be installing any colcon deps of this package from master also. I think `colcon-cmake` is one.

File: run_integration_test.sh [hunk 2]
```diff
@@ -8,6 +8,7 @@ rm -rf integration/v2_bundle
 rm -rf integration/v1.tar.gz
 rm -rf integration/v2.tar
 
+pip freeze > requirements.txt
```
Comment (murphm8) [inline run_integration_test.sh]: Add a comment on this explaining why we are exporting this.

File: Dockerfile [hunk 3]
```diff
@@ -14,9 +14,8 @@ RUN	rosdep update && \
 	rosdep install --from-paths integration/test_workspace --ignore-src -r -y
 
 RUN pip3 install --upgrade pip setuptools
-RUN pip3 install .
-RUN pip3 install -e git+https://github.com/colcon/colcon-core.git@master#egg=colcon-core
-RUN pip3 install -e git+https://github.com/colcon/colcon-ros-bundle.git@master#egg=colcon-ros-bundle
+RUN pip3 install -e .
```
Comment (murphm8) [inline Dockerfile]: Should execute this after installing from requirements, I think.

File: tox.ini (hunk 1 above)
Comment (lichenbo) [reply tox.ini]: I added the other colcon dependencies being used from github master branch

File: Dockerfile (hunk 3 above)
Comment (lichenbo) [reply Dockerfile]: Yeah, it's better

**Pull request merged**